src/uint256.h

```diff
@@ -88,12 +91,30 @@ class base_blob
     }
 };
 
+namespace detail {
+/**
+ * Writes the hex string into a new uintN_t object and only returns a value
+ * iff all of the checks pass:
+ *   - All characters are hex
+ *   - Input length is uintN_t::size()*2
+ */
+template <class uintN_t>
+std::optional<uintN_t> FromHex(std::string_view str)
+{
+    if (uintN_t::size() * 2 != str.size() || !IsHex(str)) return std::nullopt;
+    uintN_t rv;
+    rv.SetHexDeprecated(str);
+    return rv;
+}
+} // namespace detail
```

Comment:
Curious: Why introduce this namespace instead of making a protected method in `base_blob` as I did in my exploration (404c115ea0f53785a640818c2058ef0591c8c6ac)?

Reply:
Hm.. yeah, to me the namespace itself is a cognitive speed-bump but I see your point about keeping to the public members.